src/app/clusters/camera-av-stream-management-server/camera-av-stream-management-server.cpp

```diff
@@ -332,6 +332,115 @@ CHIP_ERROR CameraAVStreamMgmtServer::RemoveSnapshotStream(uint16_t snapshotStrea
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR CameraAVStreamMgmtServer::UpdateVideoStreamRefCount(uint16_t videoStreamId, bool shouldIncrement)
+{
+    auto it = std::find_if(mAllocatedVideoStreams.begin(), mAllocatedVideoStreams.end(),
+                           [videoStreamId](const VideoStreamStruct & vStream) { return vStream.videoStreamID == videoStreamId; });
+
+    if (it == mAllocatedVideoStreams.end())
+    {
+        return CHIP_ERROR_NOT_FOUND;
+    }
+
+    if (shouldIncrement)
+    {
+        if (it->referenceCount < UINT8_MAX)
+        {
+            it->referenceCount++;
+        }
+        else
+        {
+            ChipLogError(Camera, "Attempted to increment video stream %u ref count beyond max limit", videoStreamId);
+        }
+    }
+    else
+    {
+        if (it->referenceCount > 0)
+        {
+            it->referenceCount--;
+        }
+        else
+        {
+            ChipLogError(Camera, "Attempted to decrement video stream %u ref count when it was already 0", videoStreamId);
+        }
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR CameraAVStreamMgmtServer::UpdateAudioStreamRefCount(uint16_t audioStreamId, bool shouldIncrement)
+{
+    auto it = std::find_if(mAllocatedAudioStreams.begin(), mAllocatedAudioStreams.end(),
+                           [audioStreamId](const AudioStreamStruct & aStream) { return aStream.audioStreamID == audioStreamId; });
+
+    if (it == mAllocatedAudioStreams.end())
+    {
+        return CHIP_ERROR_NOT_FOUND;
+    }
+
+    if (shouldIncrement)
+    {
+        if (it->referenceCount < UINT8_MAX)
+        {
+            it->referenceCount++;
+        }
+        else
+        {
+            ChipLogError(Camera, "Attempted to increment audio stream %u ref count beyond max limit", audioStreamId);
+        }
+    }
+    else
+    {
+        if (it->referenceCount > 0)
+        {
+            it->referenceCount--;
+        }
+        else
+        {
+            ChipLogError(Camera, "Attempted to decrement audio stream %u ref count when it was already 0", audioStreamId);
+        }
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR CameraAVStreamMgmtServer::UpdateSnapshotStreamRefCount(uint16_t snapshotStreamId, bool shouldIncrement)
```

Comment:
What logic are we thinking for reference counts on a snapshot stream?  The CaptureSnapshot command would add a count then remove in the same flow, but there is a brief period where there is a count I suppose.